### ISSUE TITLE
1D Array of 4 Elements Incorrect

### ISSUE THREAD

**Issue by modaresimr** [seung-lab/connected-components-3d#103]
An error occurs when using an array of 4 elements
```python
import numpy as np
import cc3d
labels, seg_count = cc3d.connected_components(np.array([[[1,1,1,1]]]), return_N=True)
labels
```
output:
```
[1,0,0,0]
```

expected output:
```
[1,1,1,1]
```

**Comment by william-silversmith** [comment]
Thank you for reporting this! I'll take a look.

**Comment by william-silversmith** [comment]
This is due to faulty special handling of single rows of foreground for fast processing. It gets confused between the different orientations of arrays. Will issue a fix soon.

**Comment by modaresimr** [comment]
Thank you :-)

**Comment by william-silversmith** [comment]
Well, that was a subtle bug. 1D arrays are both C and F contiguous. Some parts of the code were checking for `if F else` and others were doing `if C else` and this caused 1D arrays to have problems.

**Comment by william-silversmith** [comment]
Released a fix in 3.10.3. Check it out!
